run-clang-format.py

```diff
@@ -31,13 +31,26 @@
 
 
 DEFAULT_EXTENSIONS = 'c,h,C,H,cpp,hpp,cc,hh,c++,h++,cxx,hxx'
+DEFAULT_CLANG_FORMAT_IGNORE= '.clang-format-ignore'
 
 
 class ExitStatus:
     SUCCESS = 0
     DIFF = 1
     TROUBLE = 2
 
+def get_clang_format_ignore_excludes(clang_format_file):
+    excludes = []
+    try:
+        with io.open(clang_format_file, 'r', encoding='utf-8') as clang_ignore:
+            for pattern in clang_ignore.readlines():
```

Comment:
I think you can iterate on `clang_ignore` and you will get lines out of it, so no need to call `readlines()`.